crates/storage/db/src/lib.rs

```diff
@@ -87,3 +87,89 @@ pub use abstraction::*;
 pub use reth_interfaces::db::DatabaseError;
 pub use tables::*;
 pub use utils::is_database_empty;
+
+#[cfg(feature = "mdbx")]
+use mdbx::{Env, EnvKind, WriteMap};
+
+#[cfg(feature = "mdbx")]
+/// Alias type for the database engine in use.
+pub type DatabaseEngine = Env<WriteMap>;
```

Comment:
This mixes engine with `Env`,
what about `DatabaseEnv` and `DatabaseEnvRO` or something?

now we have an alias to express a writabel DB

do we need something for `NoWriteMap` as well?

Reply:
> not sure if such differentiation is required in other databases

if this is no longer the case we can easily rename it, but now I find it fitting